test/framework/src/main/java/org/opensearch/test/rest/OpenSearchRestTestCase.java

```diff
@@ -652,6 +652,9 @@ protected static void wipeDataStreams() throws IOException {
                 for (Object snapshot : snapshots) {
                     Map<?, ?> snapshotInfo = (Map<?, ?>) snapshot;
                     String name = (String) snapshotInfo.get("snapshot");
+                    if (name.startsWith("force_preserve")) {
```

Comment:
Parallel test jobs seem to create an issue with the `preserveSnapshotsUponCompletion` flag thats set within the subclasses. Since the snapshots are shared within the cluster, some other parallel run tends to delete all the snapshots within the repository and the cluster.

I tried sequencing the job, reducing the `maxParallelForks` to 1 for the `rollingClusterUpgrade` tests but the sequence/parallelism of tests still seems to clean up the repo for the further versions.

Reply:
I know this will be in 2.0 only but for let's just go ahead and wrap in a version check and throw a javadoc so we know what this is doing

```suggestion
                    // Parallel test jobs create an issue with preserveSnapshotsUponCompletion set within the subclasses. Since the snapshots are shared within the cluster, another parallel run may delete all snapshots in the repository.
                    // For now we hack to prevent deletion of snapshots prefixed with "force_preserve"  
                    if (Version.CURRENT.onOrBefore(Version.V_2_0_0) == true && name.startsWith("force_preserve")) {
```

Reply:
Added the suggestions in.